### PR TITLE
Adds support for Drop types.

### PR DESCRIPTION
**Pull request by remexre**

### CODE REVIEW

**Comment by Luthaf** [comment]
Could you elaborate a bit on why forwarding to `write` for each field (i.e. `#(self.#fields_names_1.write(val.#fields_names_2); )*`) is not sufficient and why this needs to use ManuallyDrop?

**Comment by remexre** [comment]
Without `ManuallyDrop`, the original value still gets dropped at the end of the scope, leading to e.g. double-frees.

**Comment by Luthaf** [comment]
And why not use `std::mem::forget`?

**Comment by remexre** [comment]
Either one could work.

**Comment by Luthaf** [comment]
Ok, I hope I'll have time to do a full review of this (in particular checking that the manually drop/forget was added to all required functions) later this week.

**Comment by Luthaf** [comment]
Just had the time to finish the review here. I also rebased on top of master and added another test for ptr::write.

The code looks sound, so if you are happy with the current state I can merge and release this!

File: soa-derive-internal/src/vec.rs [hunk 1]
```diff
@@ -153,7 +160,13 @@ pub fn derive(input: &Input) -> TokenStream {
             #[doc = #vec_name_str]
             /// ::insert()`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.insert).
             pub fn insert(&mut self, index: usize, element: #name) {
-                #(self.#fields_names.insert(index, element.#fields_names);)*
+                // similar to push, we can not use move and have to rely on ptr read/write
+                unsafe {
+                    #(self.#fields_names.insert(index, ::std::ptr::read(&element.#fields_names));)*
```
Comment (remexre) [inline soa-derive-internal/src/vec.rs:165]: I think ManuallyDrop might be necessary here to be panic-safe; if I pass an out-of-bounds index, the panic should free both `element` and `std::ptr::read(&element.#fields_names)`, I think. I haven't tested that though, I'm travelling today so I should be able to give this a harder look tomorrow or the coming week. If this is the case, I'll add a regression test for it though!

Comment (Luthaf) [reply soa-derive-internal/src/vec.rs:165]: Right, I did not think about panic safety, and this is a great argument. Needs a test indeed!

Comment (Luthaf) [reply soa-derive-internal/src/vec.rs:254]: Quite a while later, I went with doing a bounds check earlier rather than later

**Comment by Luthaf** [comment]
Thanks for this PR @remexre, and sorry for the long review/update time!

**Pull request merged**